<!DOCTYPE html>
<html lang="en" dir="auto">

<head><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>WGS群体分析（一）call SNP | Sen</title>
<meta name="keywords" content="" />
<meta name="description" content="Pre-release! 在上一篇中准备好了需要的二代测序数据，这篇就开始call SNP
bwa比对 构建索引 在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。
# samtools conda install -c bioconda samtools # picard wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar java -jar picar.jar 构建索引的流程也很简单bwa index生成.amb，.ann，.bwt，.pac和.sa文件，samtools faidx生成.fai，picard CreateSequenceDictionary生成.dict
sentieon bwa index ${GENOME} samtools faidx ${GENOME} java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT} 比对与排序 这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M
( sentieon bwa mem -M -R &quot;@RG\tID:${RGID}\tSM:${SM}\tPL:${PL}&quot; -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n &#39;error&#39; ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -  统计信息 去除重复 Haplotyper ">
<meta name="author" content="">
<link rel="canonical" href="https://blog.caiisen.top/posts/wgs-2/" />
<link crossorigin="anonymous" href="/assets/css/stylesheet.min.fe63ae84f306727c713665eb03a6c71ec6c611434c8182ba17ba7e7d26a979e9.css" integrity="sha256-/mOuhPMGcnxxNmXrA6bHHsbGEUNMgYK6F7p&#43;fSapeek=" rel="preload stylesheet" as="style">
<script defer crossorigin="anonymous" src="/assets/js/highlight.min.7680afc38aa6b15ddf158a4f3780b7b1f7dde7e91d26f073e6229bb7a0793c92.js" integrity="sha256-doCvw4qmsV3fFYpPN4C3sffd5&#43;kdJvBz5iKbt6B5PJI="
    onload="hljs.initHighlightingOnLoad();"></script>
<link rel="icon" href="https://blog.caiisen.top/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://blog.caiisen.top/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://blog.caiisen.top/favicon-32x32.png">
<link rel="apple-touch-icon" href="https://blog.caiisen.top/apple-touch-icon.png">
<link rel="mask-icon" href="https://blog.caiisen.top/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<meta name="generator" content="Hugo 0.82.1" />
<meta property="og:title" content="WGS群体分析（一）call SNP" />
<meta property="og:description" content="Pre-release! 在上一篇中准备好了需要的二代测序数据，这篇就开始call SNP
bwa比对 构建索引 在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。
# samtools conda install -c bioconda samtools # picard wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar java -jar picar.jar 构建索引的流程也很简单bwa index生成.amb，.ann，.bwt，.pac和.sa文件，samtools faidx生成.fai，picard CreateSequenceDictionary生成.dict
sentieon bwa index ${GENOME} samtools faidx ${GENOME} java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT} 比对与排序 这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M
( sentieon bwa mem -M -R &quot;@RG\tID:${RGID}\tSM:${SM}\tPL:${PL}&quot; -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n &#39;error&#39; ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -  统计信息 去除重复 Haplotyper " />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://blog.caiisen.top/posts/wgs-2/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2021-07-03T21:03:20&#43;08:00" />
<meta property="article:modified_time" content="2021-07-03T21:03:20&#43;08:00" />

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="WGS群体分析（一）call SNP"/>
<meta name="twitter:description" content="Pre-release! 在上一篇中准备好了需要的二代测序数据，这篇就开始call SNP
bwa比对 构建索引 在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。
# samtools conda install -c bioconda samtools # picard wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar java -jar picar.jar 构建索引的流程也很简单bwa index生成.amb，.ann，.bwt，.pac和.sa文件，samtools faidx生成.fai，picard CreateSequenceDictionary生成.dict
sentieon bwa index ${GENOME} samtools faidx ${GENOME} java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT} 比对与排序 这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M
( sentieon bwa mem -M -R &quot;@RG\tID:${RGID}\tSM:${SM}\tPL:${PL}&quot; -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n &#39;error&#39; ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -  统计信息 去除重复 Haplotyper "/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "https://blog.caiisen.top/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "WGS群体分析（一）call SNP",
      "item": "https://blog.caiisen.top/posts/wgs-2/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "WGS群体分析（一）call SNP",
  "name": "WGS群体分析（一）call SNP",
  "description": "Pre-release! 在上一篇中准备好了需要的二代测序数据，这篇就开始call SNP\nbwa比对 构建索引 在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。\n# samtools conda install -c bioconda samtools # picard wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar java -jar picar.jar 构建索引的流程也很简单bwa index生成.amb，.ann，.bwt，.pac和.sa文件，samtools faidx生成.fai，picard CreateSequenceDictionary生成.dict\nsentieon bwa index ${GENOME} samtools faidx ${GENOME} java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT} 比对与排序 这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M\n( sentieon bwa mem -M -R \u0026quot;@RG\\tID:${RGID}\\tSM:${SM}\\tPL:${PL}\u0026quot; -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n 'error' ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -  统计信息 去除重复 Haplotyper ",
  "keywords": [
    
  ],
  "articleBody": "Pre-release! 在上一篇中准备好了需要的二代测序数据，这篇就开始call SNP\nbwa比对 构建索引 在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。\n# samtools conda install -c bioconda samtools # picard wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar java -jar picar.jar 构建索引的流程也很简单bwa index生成.amb，.ann，.bwt，.pac和.sa文件，samtools faidx生成.fai，picard CreateSequenceDictionary生成.dict\nsentieon bwa index ${GENOME} samtools faidx ${GENOME} java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT} 比对与排序 这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M\n( sentieon bwa mem -M -R \"@RG\\tID:${RGID}\\tSM:${SM}\\tPL:${PL}\" -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n 'error' ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -  统计信息 去除重复 Haplotyper ",
  "wordCount" : "76",
  "inLanguage": "en",
  "datePublished": "2021-07-03T21:03:20+08:00",
  "dateModified": "2021-07-03T21:03:20+08:00",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://blog.caiisen.top/posts/wgs-2/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Sen",
    "logo": {
      "@type": "ImageObject",
      "url": "https://blog.caiisen.top/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>
<noscript>
    <style type="text/css">
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: #1d1e20;
                --entry: #2e2e33;
                --primary: rgba(255, 255, 255, 0.84);
                --secondary: rgba(255, 255, 255, 0.56);
                --tertiary: rgba(255, 255, 255, 0.16);
                --content: rgba(255, 255, 255, 0.74);
                --hljs-bg: #2e2e33;
                --code-bg: #37383e;
                --border: #333;
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="https://blog.caiisen.top/" accesskey="h" title="Sen (Alt + H)">Sen</a>
            <span class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </span>
        </div>
        <ul id="menu">
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    
    <h1 class="post-title">
      WGS群体分析（一）call SNP
    </h1>
    <div class="post-meta">July 3, 2021
</div>
  </header> 
  <div class="post-content"><h1 id="pre-release"><em>Pre-release!</em><a hidden class="anchor" aria-hidden="true" href="#pre-release">#</a></h1>
<p>在<a href="https://blog.caiisen.top/posts/wgs-1/">上一篇</a>中准备好了需要的二代测序数据，这篇就开始call SNP</p>
<h1 id="bwa比对">bwa比对<a hidden class="anchor" aria-hidden="true" href="#bwa比对">#</a></h1>
<h2 id="构建索引">构建索引<a hidden class="anchor" aria-hidden="true" href="#构建索引">#</a></h2>
<p>在开始比对之前要先对参考基因组构建索引。需要使用到三个软件sentioen bwa（目前看来跟bwa本身没有太多区别），samtools和picard，这里主要介绍samtools和picar的安装。</p>
<pre><code># samtools
conda install -c bioconda samtools
# picard
wget https://github.com/broadinstitute/picard/releases/download/2.25.6/picard.jar
java -jar picar.jar
</code></pre><p>构建索引的流程也很简单<code>bwa index</code>生成<code>.amb</code>，<code>.ann</code>，<code>.bwt</code>，<code>.pac</code>和<code>.sa</code>文件，<code>samtools faidx</code>生成<code>.fai</code>，<code>picard CreateSequenceDictionary</code>生成<code>.dict</code></p>
<pre><code>sentieon bwa index ${GENOME}
samtools faidx ${GENOME}
java -jar picard.jar CreateSequenceDictionary REFERENCE=${GENOME} OUTPUT=${DICT}
</code></pre><h2 id="比对与排序">比对与排序<a hidden class="anchor" aria-hidden="true" href="#比对与排序">#</a></h2>
<p>这里将比对和排序两个步骤合并在一起执行。比对的部分还是使用的sentioen bwa，算法部分使用的是mem，-M</p>
<pre><code>( sentieon bwa mem -M -R &quot;@RG\tID:${RGID}\tSM:${SM}\tPL:${PL}&quot; -t ${NT} -K 10000000 ${GENOME} ${FQ1} ${FQ2} || echo -n 'error' ) | sentieon util sort -r ${GENOME} -o ${SORTED.BAM} -t ${NT} --sam2bam -i -
</code></pre>
<h1 id="统计信息">统计信息<a hidden class="anchor" aria-hidden="true" href="#统计信息">#</a></h1>
<h1 id="去除重复">去除重复<a hidden class="anchor" aria-hidden="true" href="#去除重复">#</a></h1>
<h1 id="haplotyper">Haplotyper<a hidden class="anchor" aria-hidden="true" href="#haplotyper">#</a></h1>


  </div>
  <footer class="post-footer">
  </footer>
</article>
    </main>
    <footer class="footer">
    <span>&copy; 2021 <a href="https://blog.caiisen.top/">Sen</a></span>
    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://git.io/hugopapermod" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)">
    <button class="top-link" id="top-link" type="button" accesskey="g">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
            <path d="M12 6H0l6-6z" />
        </svg>
    </button>
</a>

<script>
    let menu = document.getElementById('menu')
    menu.scrollLeft = localStorage.getItem("menu-scroll-position");
    menu.onscroll = function () {
        localStorage.setItem("menu-scroll-position", menu.scrollLeft);
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
</body>

</html>
